x-pack/legacy/plugins/siem/public/components/ml/score/anomaly_score.tsx

```diff
@@ -31,7 +31,7 @@ Icon.displayName = 'Icon';
 
 export const AnomalyScore = React.memo<Args>(
   ({ jobKey, startDate, endDate, index = 0, score, interval, narrowDateRange }): JSX.Element => {
-    const [isOpen, setIsOpen] = useState(false);
+    const [isOpen, setIsOpen] = useState<boolean>(false);
```

Comment:
When you pass in something other than `null` or `undefined` such as a `false` or an object instance to `useState`, the types are smart enough to derive them from what is passed in so that they automatically understand what the type is:

<img width="365" alt="Screen Shot 2019-09-24 at 7 46 18 PM" src="https://user-images.githubusercontent.com/1151048/65562532-00ee6f00-df04-11e9-8c97-1e64de7fab39.png">

Changes like this to the types isn't going to add stronger types to the code. I wouldn't be adding additional type templates to sections such as this. It's just a style change and not really adding anything else.

We usually remove types from where they are derived such as this example.

Reply:
ok, i just was copying what i saw elsewhere. should we not be typing the hooks unless they could be null?